pep-0684.rst

```diff
@@ -0,0 +1,640 @@
+* encourages more consistency in how runtime state is used
+* makes multiple-interpreter behavior more reliable
+* leads to fixes for long-standing runtime bugs that otherwise
+  haven't been prioritized
+* exposes (and inspires fixes for) previously unknown runtime bugs
+* facilitates cleaner runtime initialization and finalization
+* makes it easier to discover/identify CPython's runtime state
+* makes it easier to statically allocate runtime state in a consistent way
+* better memory locality for runtime state
+* structural layering of the C-API (e.g. ``Include/internal``)
+
+Furthermore, much of that work benefits other CPython-related projects:
+
+* performance improvements ("faster-cpython")
+* pre-fork application deployment (e.g. Instagram)
+* extension module isolation (see :pep:`630`, etc.)
+* embedding CPython
+
+Scale of Work
+-------------
+
+The number of global variables to be moved is large enough to matter,
+but most are Python objects that can be dealt with in large groups
+(like ``Py_IDENTIFIER``).  In nearly all cases, moving these globals
+to the interpreter is highly mechanical.  That doesn't require
+cleverness but instead requires someone to put in the time.
+
+State To Be Moved
+-----------------
+
+The remaining global variables can be categorized as follows::
+
+    global objects
+        static types (incl. exception types)
+        non-static types (incl. heap types, structseq types)
+        singletons (static)
+        singletons (initialized once)
+        cached objects
+    non-objects
+        will not (or unlikely to) change after init
+        only used in the main thread
+        initialized lazily
+        pre-allocated buffers
+        state
+
+Those globals are spread between the core runtime, the builtin modules,
+and the stdlib extension modules.
+
+For a breakdown of the remaining globals, run::
+
+    ./python Tools/c-analyzer/table-file.py Tools/c-analyzer/cpython/globals-to-fix.tsv
+
+Already Completed Work
+----------------------
+
+As mentioned, this work has been going on for many years.  Here are some
+of the things that have already been done.
+
+* cleanup of runtime initialization (see :pep:`432` / :pep:`587`)
+* extension module isolation machinery (see :pep:`384` / :pep:`3121` / :pep:`489`)
+* isolation for many builtin modules
+* isolation for many stdlib extension modules
+* addition of ``_PyRuntimeState``
+* no more ``_Py_IDENTIFIER()``
+* statically allocated:
+
+  * empty string
+  * string literals
+  * identifiers
+  * latin-1 strings
+  * length-1 bytes
+  * empty tuple
+
+Tooling
+-------
+
+As already indicated, there are several tools to help identify the
+globals and reason about them.
+
+* Tools/c-analyzer/cpython/globals-to-fix.tsv - the list of remaining globals
+* Tools/c-analyzer/c-analyzer.py
+
+  + analyze - identify all the globals
+  + check - fail if there are any unsupported globals that aren't ignored
+
+* Tools/c-analyzer/table-file.py - summarize the known globals
```

Comment:
```suggestion
* ``Tools/c-analyzer/cpython/globals-to-fix.tsv`` - the list of remaining globals
* ``Tools/c-analyzer/c-analyzer.py``
  * ``analyze`` - identify all the globals
  * ``check`` - fail if there are any unsupported globals that aren't ignored
* ``Tools/c-analyzer/table-file.py`` - summarize the known globals
```

Use code/verbatim formatting and consistent list spacing

Reply:
Ah, I misunderstood PEP 12.